Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift

```diff
@@ -0,0 +1,141 @@
+// swiftlint:disable force_try
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupportTesting
+import XCTest
+
+@testable import TuistKit
+@testable import TuistSupport
+
+public enum Destination {
+    case simulator, device
+}
+
+open class TuistAcceptanceTestCase: TuistTestCase {
+    public var xcodeprojPath: AbsolutePath!
+    public var workspacePath: AbsolutePath!
+    public var fixturePath: AbsolutePath!
+    public var derivedDataPath: AbsolutePath!
+    public var cacheDirectory: AbsolutePath!
+
+    private var sourceRootPath: AbsolutePath!
+
+    override open func setUp() {
+        super.setUp()
+
+        cacheDirectory = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        derivedDataPath = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        setenv(
+            Constants.EnvironmentVariables.forceConfigCacheDirectory,
+            cacheDirectory.pathString,
+            1
+        )
+
+        sourceRootPath = try! AbsolutePath(
+            validating: ProcessInfo.processInfo.environment[
+                "TUIST_CONFIG_SRCROOT"
+            ]!
+        )
+        environment.tuistConfigVariables[
+            Constants.EnvironmentVariables.xcbeautifyBinaryPath
+        ] = sourceRootPath
+            .appending(components: ["vendor", ".build", "debug", "xcbeautify"])
+            .pathString
+
+        DeveloperEnvironment.shared = DeveloperEnvironment()
+    }
+
+    override open func tearDown() async throws {
+        xcodeprojPath = nil
+        workspacePath = nil
+        fixturePath = nil
+        derivedDataPath = nil
+        cacheDirectory = nil
+
+        try await super.tearDown()
+    }
+
+    public func setUpFixture(_ fixture: String) throws {
+        let fixturesPath = sourceRootPath
+            .appending(component: "fixtures")
+
+        fixturePath = FileHandler.shared.currentPath.appending(component: fixture)
+
+        try FileHandler.shared.copy(
+            from: fixturesPath.appending(component: fixture),
+            to: fixturePath
+        )
+
+        try poisonCache()
+    }
+
+    /// We need to poison the cache when we start a given test as we share
+    private func poisonCache() throws {
+        let filePath = try XCTUnwrap(FileHandler.shared.glob(fixturePath, glob: "**/*.swift").first)
+        var contents = try FileHandler.shared.readTextFile(filePath)
+        contents += "\n"
+        try FileHandler.shared.write(contents, path: filePath, atomically: true)
+    }
```

Comment:
I don't understand why this is necessary.

Reply:
It's not anymore since we don't share the cache directory.

Reply:
It's not anymore 👍 I added it under the impression we would need to share the cache directory but that's actually not true.